app/ante/min_gas_filter.go

```diff
@@ -0,0 +1,40 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+)
+
+var _ sdk.AnteDecorator = EvmMinGasFilter{}
+
+type EVMKeeper interface {
+	GetParams(ctx sdk.Context) evmtypes.Params
+}
+
+type EvmMinGasFilter struct {
+	evmKeeper EVMKeeper
+}
+
+func NewEvmMinGasFilter(evmKeeper EVMKeeper) EvmMinGasFilter {
+	return EvmMinGasFilter{
+		evmKeeper: evmKeeper,
+	}
+}
+
+func (emgf EvmMinGasFilter) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	evmDenom := emgf.evmKeeper.GetParams(ctx).EvmDenom
+
+	if ctx.MinGasPrices().AmountOf(evmDenom).IsPositive() {
```

Comment:
@rhuairahrighairidh is this the best method to filter a DecCoins?  `Add` is probably inefficient, but does result in a valid sorted DecCoins at the end.

Reply:
I don't know of any much better method. I've used `Add` as well for those reasons.